core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -164,6 +172,14 @@ object RequestChannel extends Logging {
         m.totalTimeHist.update(totalTime)
       }
 
+      // Record network handler thread usage. This is included towards the request quota for
+      // the user/client. Throttling is only performed when request handler thread usage
+      // is recorded, just before responses are queued for delivery.
+      recordNetworkThreadUsageCallback match {
+        case Some(record) => record(math.max(endTimeNanos - responseDequeueTimeNanos, 0))
```

Comment:
That seems fine for now. Perhaps we could add a comment in case we need to revisit in the future?